File: ContestOfSwift/TestsOfSwift/Challenges/Challenge10Tests.swift
//
//  Challenge10Tests.swift
//  TestsOfSwift
//
//  Created by Sergio Daniel on 4/29/24.
//

import Foundation

import XCTest

final class Challenge10Tests: XCTestCase {

    // Natural to Morse

    func testTransformToMorseWithNormalString() {
        let result = Challenge10.transformToMorse(text: "hello")
        let expected = ".... . .—.. .—.. ———"
        XCTAssertEqual(result, expected)
    }

    func testTransformToMorseWithEmptyString() {
        let result = Challenge10.transformToMorse(text: "")
        let expected = ""
        XCTAssertEqual(result, expected)
    }

    func testTransformToMorseWithNumbers() {
        let result = Challenge10.transformToMorse(text: "123")
        let expected = ".———— ..——— ...——"  // Assuming your Morse code function handles numbers
        XCTAssertEqual(result, expected)
    }

    func testTransformToMorseWithSpecialCharacters() {
        let result = Challenge10.transformToMorse(text: "hello!")
        let expected = ".... . .—.. .—.. ——— —.—.——"  // Assuming '!' translates to '—.—.——'
        XCTAssertEqual(result, expected)
    }

    func testTransformToMorseWithSpaces() {
        let result = Challenge10.transformToMorse(text: "hi there")
        let expected = ".... ..  — .... . .—. ."
        XCTAssertEqual(result, expected)
    }

    // Morse to Natural

    func testTransformToNaturalWithNormalString() {
        let result = Challenge10.transformFromMorse(".... . .—.. .—.. ———")
        let expected = "HELLO"
        XCTAssertEqual(result, expected)
    }

    func testTransformToNaturalWithEmptyString() {
        let result = Challenge10.transformFromMorse("")
        let expected = ""
        XCTAssertEqual(result, expected)
    }

    func testTransformToNaturalWithNumbers() {
        let result = Challenge10.transformFromMorse(".———— ..——— ...——")
        let expected = "123" // Assuming your Morse code function handles numbers
        XCTAssertEqual(result, expected)
    }

    func testTransformToNaturalWithSpecialCharacters() {
        let result = Challenge10.transformFromMorse(".... . .—.. .—.. ——— —.—.——")
        let expected = "HELLO!" // Assuming '!' translates to '—.—.——'
        XCTAssertEqual(result, expected)
    }

    func testTransformToNaturalWithSpaces() {
        let result = Challenge10.transformFromMorse(".... ..  — .... . .—. .")
        let expected = "HI THERE"
        XCTAssertEqual(result, expected)
    }

}

File: ContestOfSharp/TestOfSharp/Challenges/Challenge10Tests.cs
namespace TestOfSharp.Challenges;

using static ContestOfSharp.Challenges.Challenge10;

[TestClass]
public class Challenge10Tests
{
    [TestMethod]
    public void TestTransformToMorse_WithNormalString()
    {
        var result = ToMorse("hello");
        var expected = ".... . .—.. .—.. ———";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToMorse_WithEmptyString()
    {
        var result = ToMorse("");
        var expected = "";
        Assert.AreEqual(expected, result);
    }

    [TestMethod]
    public void TestTransformToMorse_WithNumbers()
    {
        var result = ToMorse("123");
        var expected = ".———— ..——— ...——";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToMorse_WithSpecialCharacters()
    {
        var result = ToMorse("hello!");
        var expected = ".... . .—.. .—.. ——— —.—.——";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToMorse_WithSpaces()
    {
        var result = ToMorse("hi there");
        var expected = ".... ..  — .... . .—. .";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToNatural_WithNormalString()
    {
        var result = ToNatural(".... . .—.. .—.. ———");
        var expected = "HELLO";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToNatural_WithEmptyString()
    {
        var result = ToNatural("");
        var expected = "";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToNatural_WithNumbers()
    {
        var result = ToNatural(".———— ..——— ...——");
        var expected = "123";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToNatural_WithSpecialCharacters()
    {
        var result = ToNatural(".... . .—.. .—.. ——— —.—.——");
        var expected = "HELLO!";
        Assert.AreEqual(expected, result);
    }
    
    [TestMethod]
    public void TestTransformToNatural_WithSpaces()
    {
        var result = ToNatural(".... ..  — .... . .—. .");
        var expected = "HI THERE";
        Assert.AreEqual(expected, result);
    }
}